synapse/storage/util/id_generators.py

```diff
@@ -570,6 +574,19 @@ def _update_stream_positions_table_txn(self, txn):
         pos = (self.get_current_token_for_writer(self._instance_name),)
         txn.execute(sql, (self._stream_name, self._instance_name, pos))
 
+    def _update_stream_positions_table_conn(self, conn: LoggingDatabaseConnection):
+        # We use autocommit here so that we don't have to go through a
+        # transaction dance, which a) adds latency and b) runs the risk of
+        # serialization errors.
+        try:
+            assert hasattr(conn.conn, "autocommit")
+            conn.conn.autocommit = True  # type: ignore
+
+            with conn.cursor(txn_name="MultiWriterIdGenerator._update_table") as cur:
+                self._update_stream_positions_table_txn(cur)
+        finally:
+            conn.conn.autocommit = False  # type: ignore
```

Comment:
Should we set this back to what it was instead of always `False`?

Reply:
I mean, it really should always be `False`, but can do if you think its clearer?